azure/converters/diagnostics.go

```diff
@@ -39,11 +39,13 @@ func GetDiagnosticsProfile(diagnostics *infrav1.Diagnostics) *armcompute.Diagnos
 				},
 			}
 		case infrav1.UserManagedDiagnosticsStorage:
-			return &armcompute.DiagnosticsProfile{
-				BootDiagnostics: &armcompute.BootDiagnostics{
-					Enabled:    ptr.To(true),
-					StorageURI: &diagnostics.Boot.UserManaged.StorageAccountURI,
-				},
+			if diagnostics.Boot.UserManaged != nil {
```

Comment:
can we add a new UT case "nil diagnostics.Boot.UserManaged" to the `TestGetDiagnosticsProfile` spec under azure/converters/diagnostics_test.go?

Reply:
thank you!